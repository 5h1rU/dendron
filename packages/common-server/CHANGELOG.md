# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.3.24](https://github.com/dendronhq/dendron/compare/v0.3.23...v0.3.24) (2020-07-20)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.21](https://github.com/dendronhq/dendron/compare/v0.3.20...v0.3.21) (2020-07-19)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.20](https://github.com/dendronhq/dendron/compare/v0.3.19...v0.3.20) (2020-07-18)

### Bug Fixes

- handle special characters in file names when importing using LocalFilePod ([03e42b1](https://github.com/dendronhq/dendron/commit/03e42b167ac9f073cd56f10c4e31b5cecf66dabf))

## [0.3.19](https://github.com/dendronhq/dendron/compare/v0.3.18...v0.3.19) (2020-07-18)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.18](https://github.com/dendronhq/dendron/compare/v0.3.17...v0.3.18) (2020-07-18)

### Features

- ability to import notes from local file system [docs](https://github.com/dendronhq/dendron-template/blob/master/vault/dendron.topic.pod.md) ([e44fd11](https://github.com/dendronhq/dendron/commit/e44fd1133bec10f22831f059c8d98cf4076dcdcc))

## [0.3.17](https://github.com/dendronhq/dendron/compare/v0.3.16...v0.3.17) (2020-07-17)

### Features

- more flexible file names for [journal notes](https://github.com/dendronhq/dendron-template/blob/master/vault/dendron.special-notes.md) ([9d9f10b](https://github.com/dendronhq/dendron/commit/9d9f10bd0873c201361c4625fc49c9f62ee82991))

## [0.3.16](https://github.com/dendronhq/dendron/compare/v0.3.15...v0.3.16) (2020-07-16)

### Features

- implement journal notes. see details here: https://github.com/dendronhq/dendron-template/blob/master/vault/dendron.feature.journals.md ([5e1236f](https://github.com/dendronhq/dendron/commit/5e1236fddbf1e0fddf4c27d1a40e9841cc99974f))

## [0.3.15](https://github.com/dendronhq/dendron/compare/v0.3.14...v0.3.15) (2020-07-14)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.11](https://github.com/dendronhq/dendron/compare/v0.3.10...v0.3.11) (2020-07-13)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.10](https://github.com/dendronhq/dendron/compare/v0.1.6...v0.3.10) (2020-07-13)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.9](https://github.com/dendronhq/dendron/compare/v0.3.8...v0.3.9) (2020-07-09)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.8](https://github.com/dendronhq/dendron/compare/v0.3.7...v0.3.8) (2020-07-09)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.7](https://github.com/dendronhq/dendron/compare/v0.3.6...v0.3.7) (2020-07-08)

**Note:** Version bump only for package @dendronhq/common-server

## [0.3.6](https://github.com/dendronhq/dendron/compare/v0.3.5...v0.3.6) (2020-07-07)

**Note:** Version bump only for package @dendronhq/common-server

# [0.3.0](https://github.com/dendronhq/dendron/compare/v0.2.20...v0.3.0) (2020-07-05)

**Note:** Version bump only for package @dendronhq/common-server
