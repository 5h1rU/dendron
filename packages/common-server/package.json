{
  "name": "@dendronhq/common-server",
  "version": "0.1.6",
  "description": "common-server",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "ssh://git@github.com/dendronhq/dendron.git",
    "directory": "packages/common-server"
  },
  "author": {
    "name": "Kevin Lin",
    "email": "kevin@thence.io",
    "url": "https://kevinslin.com",
    "organization": true
  },
  "keywords": [
    ""
  ],
  "files": [
    "lib"
  ],
  "main": "lib/index",
  "types": "lib/index",
  "scripts": {
    "clean": "rimraf lib && rimraf coverage",
    "format": "",
    "lint": "",
    "prepublishOnly": "npm run build",
    "prebuild": "npm run clean && npm run format && npm run lint && echo Using TypeScript && tsc --version",
    "build": "npm run compile",
    "compile": "tsc -p tsconfig.build.json",
    "test": "jest",
    "coverage": "jest --coverage",
    "watch": "npm run compile -- --watch",
    "test:watch": "jest --watch"
  },
  "dependencies": {
    "@dendronhq/common-all": "^0.1.6",
    "gray-matter": "^4.0.2",
    "lodash": "^4.17.15",
    "minimatch": "^3.0.4",
    "pino": "^6.3.2",
    "yamljs": "^0.3.0"
  },
  "devDependencies": {
    "@types/pino": "^6.3.0",
    "@types/jest": "^23.3.3",
    "@types/lodash": "^4.14.154",
    "@types/minimatch": "^3.0.3",
    "@types/node": "^10.11.4",
    "@types/yamljs": "^0.2.31",
    "coveralls": "^3.0.2",
    "jest": "^23.6.0",
    "prettier": "^1.14.3",
    "rimraf": "^2.6.2",
    "ts-jest": "^23.10.3",
    "ts-node": "^7.0.1",
    "tslint": "^5.11.0",
    "tslint-config-prettier": "^1.15.0",
    "typescript": "^3.1.1"
  },
  "engines": {
    "node": ">=10.0.0"
  },
  "jest": {
    "preset": "ts-jest"
  },
  "gitHead": "403eb012f8b34bc115dd7dd86f80c76b39329f3b"
}
