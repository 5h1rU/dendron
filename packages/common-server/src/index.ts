export * from "./files";
export * from "./logger";
export { testUtils, FileTestUtils } from "./testUtils";
