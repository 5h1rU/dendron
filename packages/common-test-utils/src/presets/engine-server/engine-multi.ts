import { TestPresetEntry } from "../../utils";
import ENGINE_SINGLE_TEST_PRESET from "./engine-single";

const INIT = {
  WITH_STUBS: new TestPresetEntry({
    label: "with stubs",
    before: ENGINE_SINGLE_TEST_PRESET.INIT.WITH_STUBS.before,
    results: async ({}: {}) => {
      return [];
    },
  }),
  results: ENGINE_SINGLE_TEST_PRESET.INIT.WITH_STUBS.results,
};

const ENGINE_MULTI_TEST_PRESET = {
  INIT,
};

export default ENGINE_MULTI_TEST_PRESET;
