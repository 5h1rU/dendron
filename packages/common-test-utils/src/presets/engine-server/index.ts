import { ENGINE_DELETE_PRESETS } from "./delete";
import ENGINE_MULTI_TEST_PRESET from "./engine-multi";
import ENGINE_SINGLE_TEST_PRESET from "./engine-single";
import { ENGINE_GET_NOTE_BY_PATH_PRESETS } from "./getByPath";
import { ENGINE_INIT_PRESETS } from "./init";
import NOTE_REF from "./note-refs";
import { ENGINE_QUERY_PRESETS } from "./query";
import { ENGINE_RENAME_PRESETS } from "./rename";
import { ENGINE_UPDATE_PRESETS } from "./update";
import { ENGINE_WRITE_PRESETS } from "./write";

export const ENGINE_SERVER = {
  NOTE_REF,
  ENGINE_MULTI_TEST_PRESET,
  ENGINE_SINGLE_TEST_PRESET,
  ENGINE_WRITE_PRESETS,
  ENGINE_INIT_PRESETS,
  ENGINE_UPDATE_PRESETS,
  ENGINE_DELETE_PRESETS,
  ENGINE_GET_NOTE_BY_PATH_PRESETS,
  ENGINE_RENAME_PRESETS,
  ENGINE_QUERY_PRESETS,
};

export const ENGINE_PRESETS = [
  // {name: "init", presets: ENGINE_SERVER.ENGINE_INIT_PRESETS},
  // {name: "delete", presets: ENGINE_SERVER.ENGINE_DELETE_PRESETS},
  // {name: "getByPath", presets: ENGINE_SERVER.ENGINE_GET_NOTE_BY_PATH_PRESETS},
  { name: "query", presets: ENGINE_SERVER.ENGINE_QUERY_PRESETS },
  // {name: "rename", presets: ENGINE_SERVER.ENGINE_RENAME_PRESETS},
  // {name: "update", presets: ENGINE_SERVER.ENGINE_UPDATE_PRESETS},
  // {name: "write", presets: ENGINE_SERVER.ENGINE_WRITE_PRESETS},
];
