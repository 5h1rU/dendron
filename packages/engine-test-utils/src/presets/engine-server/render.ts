import { TestPresetEntryV4 } from "@dendronhq/common-test-utils";
import { ENGINE_HOOKS } from "./utils";

const NOTES = {
  BASIC: new TestPresetEntryV4(
    async ({ engine }) => {
      const { data } = await engine.renderNote({
        id: "foo",
        format: "markdown",
      });
      return [{ actual: data, expected: "<p>foo body</p>", msg: "foo" }];
    },
    {
      preSetupHook: async (opts) => {
        return ENGINE_HOOKS.setupBasic(opts);
      },
    }
  ),
};

export const ENGINE_RENDER_PRESETS = {
  NOTES,
};
