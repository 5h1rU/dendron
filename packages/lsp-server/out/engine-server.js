"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DendronEngineServer = void 0;
const engine_server_1 = require("@dendronhq/engine-server");
class DendronEngineServer {
    constructor(opts) {
        this.opts = opts;
    }
    static getOrCreate(opts) {
        if (!DendronEngineServer._instance) {
            DendronEngineServer._instance = new DendronEngineServer(opts);
        }
        return DendronEngineServer._instance;
    }
    init() {
        const root = this.opts.roots[0];
        if (!root) {
            return;
        }
        const engine = engine_server_1.DendronEngine.getOrCreateEngine({
            root: root.uri,
            forceNew: true,
        });
        this._engine = engine;
        this._engine.init();
    }
}
exports.DendronEngineServer = DendronEngineServer;
//# sourceMappingURL=engine-server.js.map