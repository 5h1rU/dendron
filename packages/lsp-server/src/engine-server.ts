import { WorkspaceFolder } from "vscode-languageserver";
import { DendronEngine } from "@dendronhq/engine-server";

type DendronEngineServerOpts = {
  roots: WorkspaceFolder[];
};

export class DendronEngineServer {
  protected _engine?: DendronEngine;

  constructor(public opts: DendronEngineServerOpts) {}

  static _instance: DendronEngineServer | null;

  static getOrCreate(opts: DendronEngineServerOpts) {
    if (!DendronEngineServer._instance) {
      DendronEngineServer._instance = new DendronEngineServer(opts);
    }
    return DendronEngineServer._instance;
  }

  init() {
    const root = this.opts.roots[0];
    if (!root) {
      return;
    }
    const engine = DendronEngine.getOrCreateEngine({
      root: root.uri,
      forceNew: true,
    }) as DendronEngine;
    this._engine = engine;
    this._engine.init();
  }
}
