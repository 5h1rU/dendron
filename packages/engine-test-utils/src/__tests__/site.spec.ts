import {
  DendronSiteConfig,
  NotePropsDictV2,
  NotePropsV2,
  NoteUtilsV2,
  WorkspaceOpts,
} from "@dendronhq/common-all";
import { tmpDir, vault2Path } from "@dendronhq/common-server";
import {
  AssertUtils,
  ENGINE_HOOKS,
  NoteTestUtilsV4,
  runEngineTestV4,
  SetupHookFunction,
} from "@dendronhq/common-test-utils";
import { createEngine, SiteUtils } from "@dendronhq/engine-server";
import _ from "lodash";
import fs from "fs-extra";

const basicSetup = (preSetupHook?: SetupHookFunction) => ({
  createEngine,
  expect,
  preSetupHook: async (opts: WorkspaceOpts) => {
    await ENGINE_HOOKS.setupBasic(opts);
    if (preSetupHook) {
      await preSetupHook(opts);
    }
  },
});

const checkNotes = (opts: {
  filteredNotes: NotePropsDictV2;
  engineNotes: NotePropsDictV2;
  match: ({
    id: string;
  } & Partial<NotePropsV2>)[];
  noMatch?: {
    id: string;
  }[];
}) => {
  const { noMatch, filteredNotes, engineNotes } = opts;
  const notesActual = _.sortBy(_.values(opts.filteredNotes), "id");
  const notesExpected = _.map(opts.match, (opts) => {
    let note = { ...engineNotes[opts.id] };
    note = { ...note, ...opts };
    return note;
  });
  expect(notesActual).toEqual(_.sortBy(notesExpected, "id"));
  if (noMatch) {
    expect(
      _.every(noMatch, ({ id }) => {
        return !_.has(filteredNotes, id);
      })
    ).toBeTruthy();
  }
};

describe("SiteUtils", () => {
  let siteRootDir: string;

  beforeEach(() => {
    siteRootDir = tmpDir().name;
  });

  describe("gen", () => {
    test("write stub", async () => {
      await runEngineTestV4(
        async ({ engine, vaults, wsRoot }) => {
          const config: DendronSiteConfig = {
            siteHierarchies: ["foo", "foobar"],
            siteRootDir,
          };
          const notes = await SiteUtils.filterByConfig({ engine, config });
          expect(notes).toMatchSnapshot();
          expect(_.size(notes)).toEqual(4);
          const vpath = vault2Path({ wsRoot, vault: vaults[0] });
          const vaultNotes = fs.readdirSync(vpath, { encoding: "utf8" });
          expect(vaultNotes).toMatchSnapshot();
          expect(
            await AssertUtils.assertInString({
              body: vaultNotes.join(" "),
              match: ["foobar.md"],
            })
          ).toBeTruthy();
        },
        {
          ...basicSetup(async (opts) => {
            const wsRoot = opts.wsRoot;
            const vault = opts.vaults[0];
            await NoteTestUtilsV4.createNote({
              fname: "foobar.ch1",
              vault,
              wsRoot,
            });
          }),
        }
      );
    });

    test("no write stub", async () => {
      await runEngineTestV4(
        async ({ engine, vaults, wsRoot }) => {
          const config: DendronSiteConfig = {
            siteHierarchies: ["foo", "foobar"],
            siteRootDir,
            writeStubs: false,
          };
          const notes = await SiteUtils.filterByConfig({ engine, config });
          expect(notes).toMatchSnapshot();
          expect(_.size(notes)).toEqual(4);
          const vpath = vault2Path({ wsRoot, vault: vaults[0] });
          const vaultNotes = fs.readdirSync(vpath, { encoding: "utf8" });
          expect(vaultNotes).toMatchSnapshot();
          expect(
            await AssertUtils.assertInString({
              body: vaultNotes.join(" "),
              nomatch: ["foobar.md"],
            })
          ).toBeTruthy();
        },
        {
          ...basicSetup(async (opts) => {
            const wsRoot = opts.wsRoot;
            const vault = opts.vaults[0];
            await NoteTestUtilsV4.createNote({
              fname: "foobar.ch1",
              vault,
              wsRoot,
            });
          }),
        }
      );
    });
  });

  describe("per note config", () => {
    test("blacklist note", async () => {
      await runEngineTestV4(
        async ({ engine }) => {
          const config: DendronSiteConfig = {
            siteHierarchies: ["foo"],
            siteRootDir,
            usePrettyRefs: true,
          };
          const notes = await SiteUtils.filterByConfig({ engine, config });
          expect(notes).toMatchSnapshot();
          expect(_.size(notes)).toEqual(1);
          checkNotes({
            filteredNotes: notes,
            engineNotes: engine.notes,
            match: [{ id: "foo", parent: null, children: [] }],
            noMatch: [{ id: "foo.ch1" }],
          });
          expect(notes["foo"].children).toEqual([]);
        },
        {
          ...basicSetup(async (opts) => {
            const wsRoot = opts.wsRoot;
            const vault = opts.vaults[0];
            return NoteTestUtilsV4.modifyNoteByPath(
              { fname: "foo.ch1", wsRoot, vault },
              (note) => {
                note.custom = { published: false };
                return note;
              }
            );
          }),
        }
      );
    });

    test("nav_exclude", async () => {
      await runEngineTestV4(
        async ({ engine }) => {
          const config: DendronSiteConfig = {
            siteHierarchies: ["foo"],
            siteRootDir,
            usePrettyRefs: true,
          };
          const notes = await SiteUtils.filterByConfig({ engine, config });
          expect(notes).toMatchSnapshot();
          expect(_.size(notes)).toEqual(2);
          checkNotes({
            filteredNotes: notes,
            engineNotes: engine.notes,
            match: [{ id: "foo", parent: null }, { id: "foo.ch1" }],
          });
        },
        {
          ...basicSetup(async (opts) => {
            const wsRoot = opts.wsRoot;
            const vault = opts.vaults[0];
            return NoteTestUtilsV4.modifyNoteByPath(
              { fname: "foo.ch1", wsRoot, vault },
              (note) => {
                note.custom = { nav_exclude: true };
                return note;
              }
            );
          }),
        }
      );
    });
  });

  describe("per hierarchy config", () => {
    let siteRootDir: string;

    beforeEach(() => {
      siteRootDir = tmpDir().name;
    });

    test("root, publish all with dup", async () => {
      await runEngineTestV4(
        async ({ engine, vaults }) => {
          const config: DendronSiteConfig = {
            siteHierarchies: ["root"],
            siteRootDir,
            usePrettyRefs: true,
            duplicateNoteBehavior: {
              action: "useVault",
              payload: {
                vault: vaults[0],
              },
            },
            config: {
              root: {
                publishByDefault: true,
              },
            },
          };
          const notes = await SiteUtils.filterByConfig({ engine, config });
          const root = NoteUtilsV2.getNoteByFnameV4({
            fname: "root",
            notes: engine.notes,
            vault: vaults[0],
          });
          checkNotes({
            filteredNotes: notes,
            engineNotes: engine.notes,
            match: [
              { id: root!.id },
              { id: "foo" },
              { id: "bar" },
              { id: "foo.ch1" },
            ],
          });
        },
        {
          createEngine,
          expect,
          preSetupHook: async (opts) => {
            await ENGINE_HOOKS.setupBasic(opts);
          },
        }
      );
    });

    test("root, publish none with dup", async () => {
      await runEngineTestV4(
        async ({ engine, vaults }) => {
          const config: DendronSiteConfig = {
            siteHierarchies: ["root"],
            siteRootDir,
            usePrettyRefs: true,
            duplicateNoteBehavior: {
              action: "useVault",
              payload: {
                vault: vaults[0],
              },
            },
            config: {
              root: {
                publishByDefault: false,
              },
            },
          };
          const notes = await SiteUtils.filterByConfig({ engine, config });
          checkNotes({
            filteredNotes: notes,
            engineNotes: engine.notes,
            match: [],
          });
        },
        {
          createEngine,
          expect,
          preSetupHook: async (opts) => {
            await ENGINE_HOOKS.setupBasic(opts);
          },
        }
      );
    });

    test("one hiearchy", async () => {
      await runEngineTestV4(
        async ({ engine }) => {
          const config: DendronSiteConfig = {
            siteHierarchies: ["foo"],
            siteRootDir,
            usePrettyRefs: true,
          };
          const notes = await SiteUtils.filterByConfig({ engine, config });
          checkNotes({
            filteredNotes: notes,
            engineNotes: engine.notes,
            match: [{ id: "foo", parent: null }, { id: "foo.ch1" }],
          });
        },
        {
          createEngine,
          expect,
          preSetupHook: ENGINE_HOOKS.setupBasic,
        }
      );
    });

    test("mult hiearchy", async () => {
      await runEngineTestV4(
        async ({ engine }) => {
          const config: DendronSiteConfig = {
            siteHierarchies: ["foo", "bar"],
            siteRootDir,
            usePrettyRefs: true,
          };
          const notes = await SiteUtils.filterByConfig({ engine, config });
          checkNotes({
            filteredNotes: notes,
            engineNotes: engine.notes,
            match: [
              { id: "foo", parent: null },
              { id: "foo.ch1" },
              { id: "bar", parent: null },
            ],
          });
        },
        {
          createEngine,
          expect,
          preSetupHook: ENGINE_HOOKS.setupBasic,
        }
      );
    });
  });
});
