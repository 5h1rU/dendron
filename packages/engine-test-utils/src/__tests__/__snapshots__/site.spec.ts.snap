// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteUtils gen no write stub 1`] = `
Object {
  "1f940238-142c-4ea1-aee0-8c315f563515": Object {
    "body": "",
    "children": Array [
      "foobar.ch1",
    ],
    "created": 1608525880209,
    "custom": Object {
      "nav_order": 1,
    },
    "data": Object {},
    "desc": "",
    "fname": "foobar",
    "id": "1f940238-142c-4ea1-aee0-8c315f563515",
    "links": Array [],
    "parent": null,
    "stub": true,
    "title": "Foobar",
    "type": "note",
    "updated": 1608525880209,
    "vault": Object {
      "fsPath": "vault1",
    },
  },
  "foo": Object {
    "body": "foo body
",
    "children": Array [
      "foo.ch1",
    ],
    "created": "1",
    "custom": Object {
      "nav_order": 0,
      "permalink": "/",
    },
    "data": Object {},
    "desc": "",
    "fname": "foo",
    "id": "foo",
    "links": Array [],
    "parent": null,
    "schema": Object {
      "moduleId": "foo",
      "schemaId": "foo",
    },
    "title": "Foo",
    "type": "note",
    "updated": "1",
    "vault": Object {
      "fsPath": "vault1",
    },
  },
  "foo.ch1": Object {
    "body": "foo.ch1 body
",
    "children": Array [],
    "created": "1",
    "custom": Object {},
    "data": Object {},
    "desc": "",
    "fname": "foo.ch1",
    "id": "foo.ch1",
    "links": Array [],
    "parent": "foo",
    "schema": Object {
      "moduleId": "foo",
      "schemaId": "ch1",
    },
    "title": "Ch1",
    "type": "note",
    "updated": "1",
    "vault": Object {
      "fsPath": "vault1",
    },
  },
  "foobar.ch1": Object {
    "body": "
",
    "children": Array [],
    "created": "1",
    "custom": Object {},
    "data": Object {},
    "desc": "",
    "fname": "foobar.ch1",
    "id": "foobar.ch1",
    "links": Array [],
    "parent": "1f940238-142c-4ea1-aee0-8c315f563515",
    "title": "Ch1",
    "type": "note",
    "updated": "1",
    "vault": Object {
      "fsPath": "vault1",
    },
  },
}
`;

exports[`SiteUtils gen no write stub 2`] = `
Array [
  "bar.md",
  "foo.ch1.md",
  "foo.md",
  "foo.schema.yml",
  "foobar.ch1.md",
  "root.md",
  "root.schema.yml",
]
`;

exports[`SiteUtils gen write stub 1`] = `
Object {
  "4b5f6cfb-8f5c-458d-a24d-44717211692b": Object {
    "body": "",
    "children": Array [
      "foobar.ch1",
    ],
    "created": 1608525880181,
    "custom": Object {
      "nav_order": 1,
    },
    "data": Object {},
    "desc": "",
    "fname": "foobar",
    "id": "4b5f6cfb-8f5c-458d-a24d-44717211692b",
    "links": Array [],
    "parent": null,
    "stub": false,
    "title": "Foobar",
    "type": "note",
    "updated": 1608525880181,
    "vault": Object {
      "fsPath": "vault1",
    },
  },
  "foo": Object {
    "body": "foo body
",
    "children": Array [
      "foo.ch1",
    ],
    "created": "1",
    "custom": Object {
      "nav_order": 0,
      "permalink": "/",
    },
    "data": Object {},
    "desc": "",
    "fname": "foo",
    "id": "foo",
    "links": Array [],
    "parent": null,
    "schema": Object {
      "moduleId": "foo",
      "schemaId": "foo",
    },
    "title": "Foo",
    "type": "note",
    "updated": "1",
    "vault": Object {
      "fsPath": "vault1",
    },
  },
  "foo.ch1": Object {
    "body": "foo.ch1 body
",
    "children": Array [],
    "created": "1",
    "custom": Object {},
    "data": Object {},
    "desc": "",
    "fname": "foo.ch1",
    "id": "foo.ch1",
    "links": Array [],
    "parent": "foo",
    "schema": Object {
      "moduleId": "foo",
      "schemaId": "ch1",
    },
    "title": "Ch1",
    "type": "note",
    "updated": "1",
    "vault": Object {
      "fsPath": "vault1",
    },
  },
  "foobar.ch1": Object {
    "body": "
",
    "children": Array [],
    "created": "1",
    "custom": Object {},
    "data": Object {},
    "desc": "",
    "fname": "foobar.ch1",
    "id": "foobar.ch1",
    "links": Array [],
    "parent": "4b5f6cfb-8f5c-458d-a24d-44717211692b",
    "title": "Ch1",
    "type": "note",
    "updated": "1",
    "vault": Object {
      "fsPath": "vault1",
    },
  },
}
`;

exports[`SiteUtils gen write stub 2`] = `
Array [
  "bar.md",
  "foo.ch1.md",
  "foo.md",
  "foo.schema.yml",
  "foobar.ch1.md",
  "foobar.md",
  "root.md",
  "root.schema.yml",
]
`;

exports[`SiteUtils per note config blacklist note 1`] = `
Object {
  "foo": Object {
    "body": "foo body
",
    "children": Array [],
    "created": "1",
    "custom": Object {
      "nav_order": 0,
      "permalink": "/",
    },
    "data": Object {},
    "desc": "",
    "fname": "foo",
    "id": "foo",
    "links": Array [],
    "parent": null,
    "schema": Object {
      "moduleId": "foo",
      "schemaId": "foo",
    },
    "title": "Foo",
    "type": "note",
    "updated": "1",
    "vault": Object {
      "fsPath": "vault1",
    },
  },
}
`;

exports[`SiteUtils per note config nav_exclude 1`] = `
Object {
  "foo": Object {
    "body": "foo body
",
    "children": Array [
      "foo.ch1",
    ],
    "created": "1",
    "custom": Object {
      "nav_order": 0,
      "permalink": "/",
    },
    "data": Object {},
    "desc": "",
    "fname": "foo",
    "id": "foo",
    "links": Array [],
    "parent": null,
    "schema": Object {
      "moduleId": "foo",
      "schemaId": "foo",
    },
    "title": "Foo",
    "type": "note",
    "updated": "1",
    "vault": Object {
      "fsPath": "vault1",
    },
  },
  "foo.ch1": Object {
    "body": "foo.ch1 body
",
    "children": Array [],
    "created": "1",
    "custom": Object {
      "nav_exclude": true,
    },
    "data": Object {},
    "desc": "",
    "fname": "foo.ch1",
    "id": "foo.ch1",
    "links": Array [],
    "parent": "foo",
    "schema": Object {
      "moduleId": "foo",
      "schemaId": "ch1",
    },
    "title": "Ch1",
    "type": "note",
    "updated": "1",
    "vault": Object {
      "fsPath": "vault1",
    },
  },
}
`;
